src/components/menu/menu-item/menu-item.component.tsx

```diff
@@ -226,10 +226,10 @@ export const MenuItem = ({
 
   const elementProps = {
     className: href || onClick ? "carbon-menu-item--has-link" : "",
-    href,
+    href: firstFocusableChild ? undefined : href,
```

Comment:
suggestion: might be worth adding some documentation around this change and the how non-interactive items will no longer be able to be navigated to via keyboard